### ISSUE TITLE
Build failure with OCaml 4.02+trunk

### ISSUE THREAD

**Issue by yminsky** [janestreet/custom_printf#1]
This prevents one from building custom_printf (and therefore Core) on the latest compiler.  The following error is from OPAM.

```
=-=-= Installing custom_printf.111.21.00 =-=-=
default    Downloading https://opam.ocaml.org/archives/custom_printf.111.21.00+opam.tar.gz
Downloading https://opam.ocaml.org/archives/custom_printf.111.21.00+opam.tar.gz
Building custom_printf.111.21.00:
  make
  make install
[ERROR] The compilation of custom_printf.111.21.00 failed.
Removing custom_printf.111.21.00.
  ocamlfind remove custom_printf

[ERROR] Due to some errors while processing custom_printf.111.21.00, the following actions will NOT proceed:
 - install core.111.21.00

===== ERROR while installing custom_printf.111.21.00 =====
# opam-version 1.1.1
# os           darwin
# command      make
# path         /Users/yminsky/.opam/4.02.0+trunk/build/custom_printf.111.21.00
# compiler     4.02.0+trunk
# exit-code    2
# env-file     /Users/yminsky/.opam/4.02.0+trunk/build/custom_printf.111.21.00/custom_printf-77337-d2d111.env
# stdout-file  /Users/yminsky/.opam/4.02.0+trunk/build/custom_printf.111.21.00/custom_printf-77337-d2d111.out
# stderr-file  /Users/yminsky/.opam/4.02.0+trunk/build/custom_printf.111.21.00/custom_printf-77337-d2d111.err
### stdout ###
# ...[truncated]
# /Users/yminsky/.opam/4.02.0+trunk/bin/ocamlfind ocamlopt -a lib/custom_printf_dummy.cmx -o lib/custom_printf.cmxa
# /Users/yminsky/.opam/4.02.0+trunk/bin/ocamlfind ocamlopt -shared lib/custom_printf.cmxa lib/custom_printf_dummy.cmx -o lib/custom_printf.cmxs
# /Users/yminsky/.opam/4.02.0+trunk/bin/ocamlfind ocamldep -syntax camlp4o -package type_conv -package camlp4.quotations -package camlp4.lib -modules syntax/pa_custom_printf.mli > syntax/pa_custom_printf.mli.depends
# /Users/yminsky/.opam/4.02.0+trunk/bin/ocamlfind ocamlc -c -g -syntax camlp4o -package type_conv -package camlp4.quotations -package camlp4.lib -I syntax -o syntax/pa_custom_printf.cmi syntax/pa_custom_printf.mli
# /Users/yminsky/.opam/4.02.0+trunk/bin/ocamlfind ocamldep -syntax camlp4o -package type_conv -package camlp4.quotations -package camlp4.lib -modules syntax/pa_custom_printf.ml > syntax/pa_custom_printf.ml.depends
# /Users/yminsky/.opam/4.02.0+trunk/bin/ocamlfind ocamlc -c -g -syntax camlp4o -package type_conv -package camlp4.quotations -package camlp4.lib -I syntax -o syntax/pa_custom_printf.cmo syntax/pa_custom_printf.ml
# + /Users/yminsky/.opam/4.02.0+trunk/bin/ocamlfind ocamlc -c -g -syntax camlp4o -package type_conv -package camlp4.quotations -package camlp4.lib -I syntax -o syntax/pa_custom_printf.cmo syntax/pa_custom_printf.ml
# File "syntax/pa_custom_printf.ml", line 104, characters 17-38:
# Error: Unbound module Printf.CamlinternalPr
# Command exited with code 2.
### stderr ###
# File "setup.ml", line 245, characters 8-26:
# Warning 3: deprecated feature: String.set
# E: Failure("Command ''/Users/yminsky/.opam/4.02.0+trunk/bin/ocamlbuild' lib/custom_printf.cma lib/custom_printf.cmxa lib/custom_printf.a lib/custom_printf.cmxs syntax/pa_custom_printf.cma syntax/pa_custom_printf.cmxa syntax/pa_custom_printf.a syntax/pa_custom_printf.cmxs sample/test.byte -use-ocamlfind -tag debug' terminated with error code 10")
# make: *** [build] Error 1
```

**Comment by agarwal** [comment]
I just ran into this too. Would be great to have this fixed, so we can start using the cool new features of 4.02.

**Comment by ghost** [comment]
It's fixed in 111.25.00:  https://github.com/ocaml/opam-repository/pull/2400

**Comment by agarwal** [comment]
Thanks!
